Ghidra/Features/Decompiler/src/main/java/ghidra/app/plugin/core/decompile/actions/ShiftPointerAction.java

```diff
@@ -0,0 +1,208 @@
+	 * @param dt the new data type
+	 * @param tool the PluginTool
+	 */
+	private void retypeSymbol(Program program, HighSymbol highSymbol, Varnode exactSpot,
+			DataType dt, PluginTool tool) {
+		HighFunction hfunction = highSymbol.getHighFunction();
+
+		boolean commitRequired = checkFullCommit(highSymbol, hfunction);
+		if (commitRequired) {
+			exactSpot = null;		// Don't try to split out if commit is required
+		}
+
+		if (exactSpot != null) { // The user pointed at a particular usage, not just the vardecl
+			try {
+				HighVariable var = hfunction.splitOutMergeGroup(exactSpot.getHigh(), exactSpot);
+				highSymbol = var.getSymbol();
+			}
+			catch (PcodeException e) {
+				Msg.showError(this, tool.getToolFrame(), "Shift Failed", e.getMessage());
+				return;
+			}
+		}
+		DataTypeManager dataTypeManager = program.getDataTypeManager();
+		boolean successfulMod = false;
+		int transaction = program.startTransaction("Shift Pointer");
+		try {
+			if (dt.getDataTypeManager() != dataTypeManager) {
+				dt = dataTypeManager.resolve(dt, null);
+			}
+			if (commitRequired) {
+				// Don't use datatypes of other parameters if the datatypes were floating.
+				// Datatypes were floating if signature source was DEFAULT
+				boolean useDataTypes =
+					hfunction.getFunction().getSignatureSource() != SourceType.DEFAULT;
+				try {
+					HighFunctionDBUtil.commitParamsToDatabase(hfunction, useDataTypes,
+						SourceType.USER_DEFINED);
+					if (useDataTypes) {
+						HighFunctionDBUtil.commitReturnToDatabase(hfunction,
+							SourceType.USER_DEFINED);
+					}
+				}
+				catch (DuplicateNameException e) {
+					throw new AssertException("Unexpected exception", e);
+				}
+				catch (InvalidInputException e) {
+					Msg.showError(this, null, "Parameter Commit Failed", e.getMessage());
+				}
+			}
+			HighFunctionDBUtil.updateDBVariable(highSymbol, null, dt, SourceType.USER_DEFINED);
+			successfulMod = true;
+		}
+		catch (DuplicateNameException e) {
+			throw new AssertException("Unexpected exception", e);
+		}
+		catch (InvalidInputException e) {
+			Msg.showError(this, tool.getToolFrame(), "Shift Failed",
+				"Failed to shift pointer '" + highSymbol.getName() + "': " + e.getMessage());
+		}
+		finally {
+			program.endTransaction(transaction, successfulMod);
+		}
+	}
+
+	/**
+	 * Retype a function return value
+	 * @param program the program
+	 * @param highFunction the function to retype
+	 * @param dt the new datatype
+	 * @param tool the PluginTool
+	 */
+	private void retypeReturn(Program program, HighFunction highFunction, DataType dt, PluginTool tool) { 
+		boolean commitRequired = checkFullCommit(null, highFunction);
+		if (commitRequired) {
+			int resp = OptionDialog.showOptionDialog(tool.getToolFrame(),
+				"Parameter Commit Required",
+				"Shifting the pointer requires all other parameters to be committed!\nContinue with shift?",
+				"Continue");
+			if (resp != OptionDialog.OPTION_ONE) {
+				return;
+			}
```

Comment:
It might not be necessary to ask the user since retyping the return type currently commits the parameters automatically anyway. This suggestion needs input from others though.

Reply:
Again, I copied that part of the code from RetypeReturnAction, but I assume I could remove it, because when shifting a local pointer, there's no such message